src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java

```diff
@@ -388,29 +388,33 @@ private static void waitFor(Supplier<Boolean> func)
     throw new TimeoutException();
   }
 
-  private static long getExecutionTimeSec(long startTime, long endTime) {
-    return (endTime - startTime) / 1000;
+  private static long getDurationFromStartMs(long startTime) {
+    final long currTime = System.currentTimeMillis();
+    return currTime - startTime;
   }
 
   void logWarningForPutAndPrecommit(long startTime, int size, String apiName) {
-    long executionTime = getExecutionTimeSec(startTime, System.currentTimeMillis());
-    if (executionTime > 300) {
+    long executionTimeMs = getDurationFromStartMs(startTime);
+    long executionTimeSec = executionTimeMs / 1000;
```

Comment:
why do we need the sec here as well?

Reply:
I retained the sec because 300 sec is more user friendly than a 300,000 milisecond timeout